### ISSUE TITLE
Using classes "defined as generic in stubs but not at runtime" as base class triggers inherit-non-class

### ISSUE THREAD

**Issue by mathiasertl** [PyCQA/pylint#4697]
`pylint>=2.9.0` triggers a false positive of `inherit-non-class` when using a class as base that was defined as a generic in a stub (or mypy plugin) but not at runtime when using the [recommended workaround from mypy](https://mypy.readthedocs.io/en/latest/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime)

### Steps to reproduce

Consider this file:

```
# pylint: disable=missing-module-docstring,missing-class-docstring
import typing

from django.db import models

if typing.TYPE_CHECKING:
    QuerySetBase = models.QuerySet[models.Model]
else:
    QuerySetBase = models.QuerySet


# class def will trigger: E0239: Inheriting 'QuerySetBase', which is not a class. (inherit-non-class)
class TestQuerySet(QuerySetBase):  
    pass
```

### Current behavior

`pylint>=2.9.0` triggers `inherit-non-class`, version `2.8.3` does not yet trigger that error.

It should be noted that a workaround is to invert the if/else statement like this:

```
if not typing.TYPE_CHECKING:
    QuerySetBase = models.QuerySet
else:
    QuerySetBase = models.QuerySet[models.Model]
```

... which will not trigger the error. Still of course, something is clearly happening that is not intended.

### Expected behavior

... not trigger that error, obviously ;-)

**Comment by Pierre-Sassoulas** [comment]
To take into account when doing #3285
